### PR TITLE
Add Shiki as an alternative to Prism

### PR DESCRIPTION
**Pull request by JuanM04**
## Changes

Adds Shiki to `@astrojs/markdown-remark`, in order to progress with [RFC 3: Replace Prism with Shiki](https://github.com/withastro/rfcs/blob/main/active-rfcs/0003-replace-prism-with-shiki.md).

## Testing

- `yarn test:match "Shiki"`

## Docs

Here is documentation about the changes: https://deploy-preview-2497--astro-docs-2.netlify.app/en/guides/markdown-content/#syntax-highlighting

### CODE REVIEW

**Review by jonathantneal — APPROVED** [review]
I would like a second look from a member who worked on the markdown component, but I like this.

Particularly packages/astro/components/Code.astro

File: packages/astro/components/Code.astro [hunk 1]
```diff
@@ -43,7 +43,7 @@ function repairShikiTheme(html: string): string {
 }
 
 const highlighter = await shiki.getHighlighter({ theme });
-const _html = highlighter.codeToHtml(code, lang);
+const _html = highlighter.codeToHtml(code, { lang });
```
Comment (matthewp) [inline packages/astro/components/Code.astro:46]: Curious why this changed. Was this breaking? Or is this an optional way to pass the lang?

Comment (natemoo-re) [reply packages/astro/components/Code.astro:46]: Looks like they are both valid. I am fine with this change given that.

https://github.com/shikijs/shiki/blob/f462aa9bc0fcacd2e86513e7efb267aaa6e17afe/packages/shiki/src/highlighter.ts#L127-L137

**Review by natemoo-re — APPROVED** [review]
Looks great! Thank you so much for the contribution!

File: packages/astro/components/Code.astro (hunk 1 above)
Comment (JuanM04) [reply packages/astro/components/Code.astro:46]: It's not breaking, it's just deprecated: https://github.com/shikijs/shiki/blob/e40647c00a036c14ae5e28dfde007f59a6485c37/packages/shiki/src/types.ts#L36

**Pull request merged**